command/daemon.go

```diff
@@ -21,36 +21,47 @@ var DaemonCmd = &cli.Command{
 	Action: daemonCommand,
 }
 
-func daemonCommand(c *cli.Context) error {
-	ctx, cancel := context.WithCancel(ProcessContext())
-	defer cancel()
+func daemonCommand(cctx *cli.Context) error {
+	// TODO: get these from config file
+	cacheSize := int(cctx.Int64("cachesize"))
+	dir := cctx.String("dir")
+	storeType := cctx.String("storage")
+	finderAddr := cctx.String("finder_ep")
+	adminAddr := cctx.String("admin_ep")
+	p2pEnabled := cctx.Bool("enablep2p")
 
 	log.Infow("Starting node deamon")
-	n, err := node.New(ctx, c)
+	n, err := node.New(cacheSize, dir, storeType, finderAddr, adminAddr, p2pEnabled)
 	if err != nil {
 		return err
 	}
 
-	go func() {
-		// Wait for SIGINT (CTRL-c), then close server and exit.
-		sigint := make(chan os.Signal, 1)
-		signal.Notify(sigint, os.Interrupt)
-		<-sigint
+	n.Start()
+
+	// Wait for SIGINT (CTRL-c), then close server and exit.
+	sigint := make(chan os.Signal, 1)
```

Comment:
This approach makes way more sense than what I was doing. I can't remember why I included this in the routine in the first place 🤦‍♂️